### ISSUE TITLE
Vanilla slowdown since 1.0.73

### ISSUE THREAD

**Issue by andiandi13** [vanilla-music/vanilla#859]
I just noticed that Vanilla is slower since the last update, particularly when opening now playing screen, I can't confirm yet if it's slower in other things.

Two videos to illustrate that :

1.0.72 (fast) :

https://youtu.be/oIYBiA5XxkM

1.0.73 (slow) :

https://youtu.be/PjUFH4K-g24

I felt an unconfortable behavior before I noticed it was the app and not my phone slowing down or something.

Is the culprit the new plugin system ? Idk, but Vanilla should stay fast...

**Comment by andiandi13** [comment]
Sorry I thought it wasn't necessary to specify those things in the issue.

So yes I'm using 4.4.4 on a Galaxy S5. Same issue with nightly or play store.

**Comment by adrian-bl** [comment]
The diff between 1.0.72 and 1.0.73 is indeed only plugin changes:
https://github.com/vanilla-music/vanilla/compare/1.0.72...1.0.73

Do you actually have any plugins installed?

I suppose that the call to checkPlugins() got more expensive:
We do this in onCreateMenu which (afaik?) runs while the activity gets created on 4.x (on 5.x, it runs when you hit the menu button)

**Comment by andiandi13** [comment]
Yes I have the tag editor plugin. I deleted it and it's now faster, like before !

> I suppose that the call to checkPlugins() got more expensive:
> We do this in onCreateMenu which (afaik?) runs while the activity gets created on 4.x (on 5.x, it runs when you hit the menu button)

So is it possible to fix it for 4.x ? Or at least to use the old plugin system only for Kit Kat ?

**Comment by andiandi13** [comment]
@adrian-bl Its not fixed ?

**Comment by Adonai** [comment]
Oh! Didn't notice this bug!

Will take a look closer to weekend
